hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/JoinCheckOperation.java

```diff
@@ -29,17 +29,22 @@
 
 import java.io.IOException;
 
+import static com.hazelcast.version.Version.MAJOR_MINOR_VERSION_COMPARATOR;
+
+/**
+ * Validate whether clusters may merge to recover from a split brain, based on configuration & cluster version.
+ */
 public class JoinCheckOperation extends AbstractJoinOperation {
 
```

Comment:
Can't we name this operation SplitBrainMergeCheckOperation?
It's confusing, I thought it's invoked on normal join.